aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/additional-actions/getPropertyDependsOnAction.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @author Alexey Stsefanovich (ala'n), Liubou Masiuk (liubou-masiuk), Yana Bernatskaya (YanaBr)
+ * @version 2.2.4
+ * Custom action to get component property
+ * property path can be relative (e.g. 'node/nestedProperty' or '../../parentCompProperty')
+ *
+ * query - property path
+ * */
+
+(function (Granite, $, DependsOn) {
+    'use strict';
+
+    const PARENT_DIR_REGEX = /\.\.\//g;
+
+    function getLevel(path) {
+        const dirMatches = path.match(PARENT_DIR_REGEX);
+        return dirMatches && dirMatches.length || 0;
+    }
+
+    class RequestCache {
```

Comment:
As the RequestCache have state - it is better to use classic singleton pattern instead of store sate as static field